pkg/sentry/fsimpl/gofer/gofer.go

```diff
@@ -95,7 +95,42 @@ const (
 	cacheRemoteRevalidating  = "remote_revalidating"
 )
 
-const defaultMaxCachedDentries = 1000
+const (
+	defaultMaxCachedDentries         = 1000
+	defaultMaxCachedNegativeChildren = 1000
+)
+
+// +stateify savable
+type negativeNamesCache struct {
+	// namesList stores negative names with fifo list.
+	// name stored in namesList only means it used to be negative
+	// at the moment you pushed it to the list.
+	namesList stringList
+	namesLen  uint64
+}
+
+func (cache *negativeNamesCache) Empty() bool {
```

Comment:
But the tests are in the `gofer` package itself. So no need to export these methods.

Note that this is an unexported type, so having exported methods on this does nothing, because you can not  use this from another package anyway. Exported methods on unexported types makes sense when implementing interfaces, which is not the case here.

Reply:
Yes, I removed this function.

Reply:
The methods are still exported, i.e. they are named `Add()` and `IsInited()`, instead of `add()` and `isInited()`. The original comment was about unexporting these methods (i.e. making them start with a lower case character).